h2o-k8s/build.gradle

```diff
@@ -8,6 +8,12 @@ repositories {
 
 dependencies {
     compile project(":h2o-core")
+    compile 'org.nanohttpd:nanohttpd:2.3.1'
```

Comment:
Is there any reason why we can't implement this feature via a Jetty servlet?

Reply:
+1

Reply:
Explained in the description.

Reply:
Our jetty servlet starts a lot later and required H2O configured. Also, we don't want to let our users call the Jetty ensemble due to cloud locking issues. Also, what if the Jetty is secured by the user ? Could also happen on Kubernetes. The K8S API should stand aside.

Reply:
For more information see `gateHandler` method of `H2OHttpViewImpl` class.

This method is waiting for
```
    ServletUtils.startRequestLifecycle();
    while (! isAcceptingRequests()) {
      try { Thread.sleep(100); }
      catch (Exception ignore) {}
    }
```

which is waiting for `H2O.startServingRestApi` call. 

In order to go start the REST API, we'd need to do the preparations for H2O core beforehand (which is possible to do) and start serving the REST API sooner. The effort would then be duplicated. Also, we'd need to filter out only certain requests (probes), otherwise H2O would start accepting requests which could lock the cluster prematurely. There are several potential deadlocks.

This abstracts away K8S REST API from H2O's API and acts as a separate service (which it is).

Reply:
The different lifecycles of the endpoints and also different needs of authentication/security are IMHO good reasons to avoid relying on the embedded Jetty.

Reply:
@mn-mikke @honzasterba WDYT?

Reply:
Ok, thanks @Pscheidl for your explanation! The library seems to be light-weight and doesn't have any other dependencies. it seems that SW/H2O assembly jar won't grow much, so I'm fine with that.

Reply:
It weights only a few tens of kilobytes in total.